satpy/readers/fci_l1c_fdhsi.py

```diff
@@ -152,6 +179,23 @@ def get_dataset(self, key, info=None):
         info.pop("units")
         attrs.pop("units")
 
+        # For each channel, the effective_radiance contains in the
+        # "ancillary_variables" attribute the value "pixel_quality".  In
+        # FileYAMLReader._load_ancillary_variables, satpy will try to load
+        # "pixel_quality" but is lacking the context from what group to load
+        # it.  Until we can have multiple pixel_quality variables defined (for
+        # example, with https://github.com/pytroll/satpy/pull/1088), rewrite
+        # the ancillary variable to include the channel.  See also
+        # https://github.com/pytroll/satpy/issues/1171.
+        if attrs["ancillary_variables"] == "pixel_quality":
+            attrs["ancillary_variables"] = key.name + "_pixel_quality"
+        else:
+            raise ValueError(
+                "Unexpected value for attribute ancillary_variables, "
+                "which I intend to rewrite (see "
+                "https://github.com/pytroll/satpy/issues/1171 for why). "
```

Comment:
Maybe these two lines aren't necessary, as it's already written in the comments above ? Also, please use impersonal mode when writing error/log messages

Reply:
The comments offer a bit more explanation than the error message, but a user may see only the error message unless reading the source code, so I don't think repeating part of a comment in an error message is redundant.  

I'll change the error message although [opinions differ on first person in error messages](https://ux.stackexchange.com/questions/122299/should-software-refer-to-itself-in-the-first-person-especially-in-error-message).

Reply:
"I'm sorry Dave, I'm afraid I can't do that" ;)